service/history/queues/priority_assigner.go

```diff
@@ -47,15 +47,23 @@ func NewPriorityAssigner() PriorityAssigner {
 }
 
 func (a *priorityAssignerImpl) Assign(executable Executable) tasks.Priority {
-	switch executable.GetType() {
+	taskType := executable.GetType()
+	switch taskType {
 	case enumsspb.TASK_TYPE_DELETE_HISTORY_EVENT,
 		enumsspb.TASK_TYPE_TRANSFER_DELETE_EXECUTION,
-		enumsspb.TASK_TYPE_VISIBILITY_DELETE_EXECUTION:
+		enumsspb.TASK_TYPE_VISIBILITY_DELETE_EXECUTION,
+		enumsspb.TASK_TYPE_ARCHIVAL_ARCHIVE_EXECUTION,
+		enumsspb.TASK_TYPE_UNSPECIFIED:
 		// add more task types here if we believe it's ok to delay those tasks
 		// and assign them the same priority as throttled tasks
 		return tasks.PriorityLow
 	}
 
+	if _, ok := enumsspb.TaskType_name[int32(taskType)]; !ok {
+		// low priority for unknown task types
+		return tasks.PriorityLow
```

Comment:
just curious, is this assigner DI-able?

Reply:
Currently it's not created through DI. But each queue can use a different priority assigner when creating its queue factory. e.g. https://github.com/temporalio/temporal/blob/master/service/history/transferQueueFactory.go#L92